CIDER/cider.py

```diff
@@ -1062,55 +1215,62 @@ def lipinski_plot(
 
-    def _get_Murcko_scaffold(
+    def _get_scaffold(
         self,
         moleculeset: Chem.SDMolSupplier,
         number_of_structures: int = 5,
         structures_per_row: int = 5,
         image_size: int = 200,
         framework: bool = False,
-        skeleton: bool = False,
+        graph_framework: bool = False,
         normalize: bool = True,
-    ) -> Tuple[list, count]:  # what to use here?
+    ) -> Tuple[PIL.PngImagePlugin.PngImageFile, list, pd.core.series.Series]:
         """
-        This function creates a grid images of a chosen number of scaffolds/frameworks/cyclic skeletons for the molecules in a given SDMolSupplier Object. The scaffolds/frameworks/cyclic skeletons are sorted by their frequency.
-        The relative or absolute number of occurrence of a scaffold/framework/cyclic skeleton in the dataset is shown below each image.
+        This function creates a grid images of a chosen number of scaffolds/frameworks/graph framework for the molecules in a given SDMolSupplier Object. The scaffolds/frameworks/graph framework are sorted by their frequency.
+        The relative or absolute number of occurrence of a scaffold/framework/graph framework in the dataset is shown below each image.
 
         args:
             moleculeset (Chem.SDMolSupplier): SDMolSupplier Objects
             number_of_structures (int): Number of structures displayed in the grid images (default: 5).
             structures_per_row (int): Number of structures in every row of the grid image (default: 5).
             image_size (int): Size of the image for a single molecule (default: 200).
             framework (bool): Remove terminal atoms with double bond (default: False).
-            skeleton (bool): Creating cyclic skeleton / graph framework (default: False).
+            graph_framework (bool): Creating graph framework (default: False).
             normalize (bool): Using relative numbers for scaffold analysis (default: True).
 
         returns:
-            structure_grid (PIL.PngImageFile): Grid image with most frequent scaffolds/frameworks/cyclic skeletons.
-            structure_list (list): List of scaffolds/frameworks/cyclic skeletons for every molecule.
-            structure_counts (pandas.Series): Absolute or relative frequency of each scaffold/frameworks/cyclic skeletons.
+            structure_grid (PIL.PngImageFile): Grid image with most frequent scaffolds/frameworks/graph framework.
+            structure_list (list): List of scaffolds/frameworks/graph framework for every molecule.
+            structure_counts (pandas.Series): Absolute or relative frequency of each scaffold/frameworks/graph framework.
         """
         structure_list = []
         scaffold_list = []
         for mol in moleculeset:
-            scaffold = MurckoScaffold.GetScaffoldForMol(mol)
+            try:
+                Chem.Kekulize(mol)
+                scaffold = MurckoScaffold.GetScaffoldForMol(mol)
+            except BaseException:
```

Comment:
Is that exception you had here? Wasn't this a "KekulizationError" or something?

Reply:
Yes, the error was KekulizeException. But when I put this in I also got a warning, that the expression is not defined. Therefore, I change it to BaseException.

Reply:
And BaseException would catch the KekulizeException? Probably, it must be explicitly imported from RDKit to be able to catch it specifically.

Reply:
It will not work with the molecules that initially caused the KekulizeException, because now they work just fine due to this part. But the exception will be raised when using faulty molecules, that would normally be deleted while importing.

Reply:
I would still like to catch the KekulizeException here if possible. Can you try importing it?